### PR TITLE
Fix implementation of OnChipScanComplete and OnScanComplete

### PR DESCRIPTION
**Pull request by jlatusek**
### Problem

When the chip-tool is shutting down because of timeout from [StartWaiting](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/examples/chip-tool/commands/common/CHIPCommand.cpp#L218) then [mBleLayer->mBleTransport](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/src/transport/raw/BLE.cpp#L50) is firstly set to nullptr and [OnBleConnectionError](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/src/ble/BleLayer.cpp#L773) is called on this pointer. 

### Fix

I've changed the implementation of [OnScanComplete](https://github.com/project-chip/connectedhomeip/blob/733afaabd512cbf396d3edf09805d1054a6ec5f8/src/platform/Linux/BLEManagerImpl.cpp#L805) to do not call `OnConnectionError` it seems that this isn't the right place to do it. I've created `OnScanError`, which is called when the error occurs during scanning. Now it is called from [TimerExpiredCallback](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/src/platform/Linux/bluez/ChipDeviceScanner.cpp#L151)  which is the callback for a timeout for scan task. For [kConnectTimeout](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/src/platform/Linux/BLEManagerImpl.cpp#L799) `OnConnectionError` is called [here](https://github.com/jlatusek/connectedhomeip/blob/1e8a74435aff49ab5783f616a90b6fb99b3dea0e/src/platform/Linux/BLEManagerImpl.cpp#L268) if an error occurred.

### Testing

Manually tested on the Linux platform on one device I've run:
``` sh
 ./out/linux-x64-light/chip-lighting-app --discriminator 42 --passcode 1234 --wifi 
 ```
and on the other Linux device I've run: 
``` sh
./out/linux-x64-chip-tool/chip-tool pairing ble-wifi 0x01 <wifi_name> <wifi_pass> 1234 42
```

### CODE REVIEW

File: src/platform/Linux/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -804,15 +804,20 @@ void BLEManagerImpl::OnDeviceScanned(BluezDevice1 * device, const chip::Ble::Chi
...
+    case BleScanState::kNotScanning:
         ChipLogProgress(Ble, "Scan complete notification without an active scan.");
-        return;
+        break;
+    case BleScanState::kScanForAddress:
+    case BleScanState::kScanForDiscriminator:
+        ChipLogProgress(Ble, "Scan complete. No matching device found.");
+        break;
+    case BleScanState::kConnecting:
+        BleConnectionDelegate::OnConnectionError(mBLEScanConfig.mAppState, CHIP_ERROR_TIMEOUT);
```
Comment (Damian-Nordic) [inline src/platform/Linux/BLEManagerImpl.cpp:817]: Can you explain the change? `BleScanState::kConnecting` state is entered when a matching device is found. The scanning is stopped right after that. If think the goal of this code was to error out only if none matching device has been found, no?

**Review by andy31415 — DISMISSED** [review]
Please update the summary: the text asked you to provide details and delete the instruction text. You did not delete the instruction text.

**Review by andy31415 — COMMENTED** [review]
@jlatusek - Testing via CI is not possible: CI has no BLE capabilities. This needs at least some manual check.

File: src/platform/Linux/BLEManagerImpl.cpp (hunk 1 above)
Comment (jlatusek) [reply src/platform/Linux/BLEManagerImpl.cpp]: You are right. That was my mistake, and I've changed that.

**Review by arkq — APPROVED** [review]

**Pull request closed without merging**